src/core/cache/value.ts

```diff
@@ -0,0 +1,19 @@
+import { CacheKeyFragment, getCache } from '.';
+
+export default <TKey extends CacheKeyFragment[]>() =>
+    <TEntry>(fn: (...a: TKey) => TEntry) => {
```

Comment:
Turns out TS is great at inferring types when there's only one generic, and not so good when there are 2 or more.. in this case it makes sense to have a first call to access the cache entry and a second to execute the memoized function so it works out well enough.